### PR TITLE
[fix][test] Fix `ClientDeduplicationFailureTest.testClientDeduplicationWithBkFailure`

### PR DESCRIPTION
**Pull request by mattisonchao**
### Motivation

image

This problem may be caused by #16605.

### Modifications

- Fix `ClassCastException`

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [x] `doc-not-needed` 
(Please explain why)
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Review by Technoboy- — APPROVED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/client/api/ClientDeduplicationFailureTest.java [hunk 1]
```diff
@@ -60,7 +59,7 @@
 import org.testng.annotations.Test;
 
 @Slf4j
-@Test(groups = "quarantine")
+@Test(groups = "broker")
```
Comment (lhotari) [inline pulsar-broker/src/test/java/org/apache/pulsar/client/api/ClientDeduplicationFailureTest.java:62]: Was this added in #16605? before taking tests out of quarantine, it should be ensured that the test is not flaky any more. The problem fixed in this PR isn't about flakiness. Have you verified that ClientDeduplicationFailureTest is no more flaky?

Comment (mattisonchao) [reply pulsar-broker/src/test/java/org/apache/pulsar/client/api/ClientDeduplicationFailureTest.java:62]: How to verify? Do we have any rules for leaving the flaky test?

Comment (gaoran10) [reply pulsar-broker/src/test/java/org/apache/pulsar/client/api/ClientDeduplicationFailureTest.java:62]: Maybe we can change the test group in another PR.

@lhotari I think it's hard to judge whether the test is flaky or not, maybe the flaky problem was fixed in some PRs. Maybe we need to find a way to observe the failed frequency of the flaky tests.

Comment (mattisonchao) [reply pulsar-broker/src/test/java/org/apache/pulsar/client/api/ClientDeduplicationFailureTest.java:62]: We can discuss it in another thread. I will roll back the test group.

Comment (mattisonchao) [inline pulsar-broker/src/test/java/org/apache/pulsar/client/api/ClientDeduplicationFailureTest.java:62]:
```suggestion
@Test(groups = "quarantine")
```

**Comment by mattisonchao** [comment]
Done. Please review again @lhotari  @gaoran10

**Review by gaoran10 — APPROVED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/client/api/ClientDeduplicationFailureTest.java (hunk 1 above)
Comment (gaoran10) [reply pulsar-broker/src/test/java/org/apache/pulsar/client/api/ClientDeduplicationFailureTest.java]: Thanks, @mattisonchao, in fact, if we can't reproduce the problem, it seems it's hard to judge whether the test is flaky or not, maybe run hundreds of times, and the problem presents a few times.

**Review by lhotari — APPROVED** [review]

**Pull request merged**